shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -859,6 +921,49 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], newDefs: Bo
   
   final def exprCont(acc: Term, prec: Int, allowNewlines: Bool)(implicit et: ExpectThen, fe: FoundErr, l: Line): IfBody \/ Term = wrap(prec, s"`$acc`", allowNewlines) { l =>
     cur match {
+      case (QUOTE, l) :: _ => cur match {
+        case _ :: (KEYWORD(opStr @ "=>"), l0) :: _ if opPrec(opStr)._1 > prec =>
+          consume
+          consume
+          exprCont(Quoted(Lam(acc match {
+            case t: Tup => t
+            case _ => PlainTup(acc)
+          }, Unquoted(expr(1)(fe, implicitly)))), prec, allowNewlines)
+        case _ :: (br @ BRACKETS(Round, toks), loc) :: _ =>
+          consume
+          consume
+          val as = rec(toks, S(br.innerLoc), br.describe).concludeWith(_.argsMaybeIndented()).map {
+            case nme -> Fld(flgs, t) => nme -> Fld(flgs, Unquoted(t))
+          }
+          val res = App(Unquoted(acc), Tup(as).withLoc(S(loc)))
+          exprCont(Quoted(res), prec, allowNewlines)
+        case _ :: (IDENT(opStr, true), l0) :: _ if opPrec(opStr)._1 > prec =>
+          val currentOp = Var(opStr).withLoc(S(l0))
+          consume
+          def insQuotes(term: Term): Term = term match {
+            // * Find the correct operator and insert quotes and unquotes.
+            case App(vop @ Var(op), Tup(lhs :: rhs :: Nil)) if op === opStr && currentOp.toLoc === vop.toLoc =>
```

Comment:
NO. This is not some low-quality C++ or Java codebase. We don't do this sort of things.